Readme.md

```diff
@@ -225,134 +244,204 @@ is of interest to you. This will likely be done by your favorite IDE, probably
 collection of many papyrus scrolls.
 
+## Architecture
 
-## Legacy (MISC/OLD) documentation (Inaccurate):
-
-
-#### Generation (`papyri gen`)
+### Generation (`papyri gen`)
 
-Collects the documentation of a project into a DocBundle -- a number of
-DocBlobs (currently json files), with a defined semantic structure, and
+Collects the documentation of a project into a *DocBundle* -- a number of
+*DocBlobs* (currently json files), with a defined semantic structure, and
 some metadata (version of the project this documentation refers to, and
 potentially some other blobs).
 
-During the generation a number of normalisation and inference can and should
-happen, for example
+During the generation a number of normalisation and inference steps can and
+should happen. For example:
 
-  - using type inference into the `Examples` sections of docstrings and storing
+  - Using type inference into the `Examples` sections of docstrings and storing
     those as pairs (token, reference), so that you can later decide that
     clicking on `np.array` in an example brings you to numpy array
-    documentation; whether or not we are currently in the numpy doc.
-  - Parsing "See Also" into a well defined structure
-  - running Example to generate images for docs with images (not implemented)
-  - resolve package local references for example building numpy doc
-    "`zeroes_like`" is non ambiguous and shoudl be Normalized to
-    "`numpy.zeroes_like`", `~.pyplot.histogram`, normalized to
-    `matplotlib.pyplot.histogram` as the **target** and `histogram` as the text
-    ...etc.
+    documentation; whether or not we are currently in the numpy documentation;
+  - Parsing "See Also" into a well defined structure;
+  - Running examples to generate images for docs with images (partially
+    implemented);
+  - Resolve local references. For example, when building the NumPy docs,
+    `zeroes_like` is non-ambiguous and should be normalized to
+    `numpy.zeroes_like`. Similarly, `~.pyplot.histogram`, should be normalized
+    to `matplotlib.pyplot.histogram` as the **target** and `histogram` as the
+    text.
 
 The Generation step is likely project specific, as there might be import
-conventions that are per-project and should not need to be repeated (`import
-pandas as pd`, for example,)
+conventions that are defined per-project and should not need to be repeated
+(`import pandas as pd`, for example.)
+
+The generation step is likely to be the most time consuming, and for each
+project, results in the following outputs:
+
+- A `papyri.json` file, which is a list of unique qualified names corresponding
+  to the documented objects and some metadata;
+- A `toc.json` file, ?
+- An `assets` folder, containing all the images generated during the
+  generation;
+- A `docs` folder, ?
+- An `examples` folder, ?
+- A `module` folder, containing one json file per documented object. 
+
+After the generation step, *what should have been processed*?
```

Comment:
This is the crucial step that will help new contributors, as it will probably point people to *what hasn't been implemented yet*